### DISCUSSION TITLE
Ordering Catalog Price Rules

### DISCUSSION THREAD

**Discussion by elburgl69** [PrestaShop/PrestaShop#33409] [Ideas]
#### Is your feature request related to a problem?

The first qualifying Catalog Price Rule is applied to products. I really get frustrated recreating rules to reorder them.

#### Describe the solution you'd like

An 'order' column must be added to specific_price_rule, that allows ordering of the rules. And add the order column to AdminSpecificPriceRuleControllerCore list (in Constructor).

Make sure when retrieving specific price (SpecificPriceCore::getSpecificPrice()) it is done in order of the 'order' column ascending (in stead of id_specific_price_rule ASC). 

#### Alternatives you've considered
1. Implement a hook in getSpecificPrince() that receives the query result of all qualifying Specific Prices in order to select applicable one in the hook.
2. Messing around with id_specific_price_rule in the database since it controls the order and thus the first row of the resulting query. 

#### Additional context
ProductCore::priceCalculation() is used to calculate the product price. It calls SpecificPrice::getSpecificPrice() to get the applicable pricing rule. So adding the ordering column should be straight forward since ListHelper already supports adding ordering columns.

**Comment by hibatallahAouadni** [comment]
Hello @elburgl69 

Thanks for the suggestion, the Developer Team will take a look.
Please stay tuned !

Ping @PrestaShop/prestashop-core-developers  , it's a bit frustrating to figure out which one you want, it will be better to add an "order" column to cart rules. It will be easy to organise it :wink: do you agree with me?

**Comment by vardvaskan** [comment]
There's already a "score" for specific price rules that should pick the "most specific" rule for a customer. For example, if there's a general discount for a category of 15%, and a discount for the same category and a specific customer group of 20%, the latter should be applied. But the query in method getSpecificPrice in class SpecificPrice has ORDER BY id_specific_price_rule ASC before score DESC. Changing the order of those two should solve the problem.
